minio/minio
# Return additional user metadata in GetUserInfo API

## Description

This change stores additional metadata about the source/authentication
of a user and returns this info for builtin IDP and LDAP users in the
GetUserInfo API. This metadata will be used in other informational
user/group related APIs and track each credential stored by the server.

~This needs https://github.com/minio/minio/pull/17426 to be merged first as it is developed on top of that change.~

## Motivation and Context

Motivation here is to make credentials (accounts, STS creds, service accounts) more trackable and to expose the owner of each access key in user info APIs. In this PR, builtin IDP users and LDAP users are clearly identified.


## How to test this PR?

Needs https://github.com/minio/madmin-go/pull/201 and https://github.com/minio/mc/pull/4613

See the mc PR for example outputs with this change.


## Types of changes

## Checklist:
